ibis/backends/base/__init__.py

```diff
@@ -753,125 +749,83 @@ def _get_backend_names() -> frozenset[str]:
-    """Connect to file located at `path`."""
-    return _connect(path, **kwargs)
-
-
-@_connect.register(r".+\.(?P<backend>.+)", priority=1)
-def _(path: str, *, backend: str, **kwargs: Any) -> BaseBackend:
-    """Connect to given path.
-
-    The extension is assumed to be the name of an ibis backend.
-
-    Examples
-    --------
-    >>> con = ibis.connect("file://relative/path/to/data.duckdb")
-    """
-    return getattr(ibis, backend).connect(path, **kwargs)
-
-
-@functools.singledispatch
-def connect(resource: Path | str, **_: Any) -> BaseBackend:
-    """Connect to `resource`.
 
-    `resource` can be a `pathlib.Path` or a `str` specifying a URL or path.
-
-    Examples
-    --------
-    >>> con = ibis.connect("duckdb:///absolute/path/to/data.db")
-    >>> con = ibis.connect("relative/path/to/data.duckdb")
+    Connect to a postgres server:
+    >>> con = ibis.connect("postgres://user:password@hostname:5432")
     """
-    raise NotImplementedError(type(resource))
-
-
-@connect.register
-def _(path: Path, **kwargs: Any) -> BaseBackend:
-    return _connect(str(path), **kwargs)
-
-
-@connect.register
-def _(url: str, **kwargs: Any) -> BaseBackend:
-    return _connect(url, **kwargs)
-
-
-@_connect.register(
-    r"(?P<backend>.+)://(?P<filename>.+\.(?P<extension>.+))",
-    priority=11,
-)
-def _(
-    _: str,
-    *,
-    backend: str,
-    filename: str,
-    extension: str,
-    **kwargs: Any,
-) -> BaseBackend:
-    """Connect to `backend` and register a file.
-
-    The extension of the file will be used to register the file with
-    the backend.
-
-    Examples
-    --------
-    >>> con = ibis.connect("duckdb://relative/path/to/data.csv")
-    >>> con = ibis.connect("duckdb:///absolute/path/to/more/data.parquet")
-    """
-    con = getattr(ibis, backend).connect(**kwargs)
-    con.register(f"{extension}://{filename}")
-    return con
-
-
-@_connect.register(r".+\.(?:parquet|csv(?:\.gz)?)", priority=8)
-def _(filename: str, **_: Any) -> BaseBackend:
-    """Connect to the default backend (DuckDB) and return a table.
-
-    Examples
-    --------
-    >>> table1 = ibis.connect("relative/path/to/file1.csv")
-    >>> table2 = ibis.connect("relative/path/to/file2.parquet")
-    >>> table3 = ibis.connect("relative/path/to/file3.csv.gz")
-    """
-    con = ibis.duckdb.connect()
-    return con.register(filename)
+    url = resource = str(resource)
+
+    if re.match("[A-Za-z]:", url):
+        # windows path with drive, treat it as a file
+        url = f"file://{url}"
+
+    parsed = urllib.parse.urlparse(url)
+    scheme = parsed.scheme or "file"
+
+    # Merge explicit kwargs with query string, explicit kwargs
+    # taking precedence
+    kwargs = dict(urllib.parse.parse_qsl(parsed.query), **kwargs)
+
+    if scheme == "file":
+        path = parsed.netloc + parsed.path
+        if path.endswith(".duckdb"):
+            return ibis.duckdb.connect(path, **kwargs)
+        elif path.endswith(".sqlite"):
```

Comment:
Can we add `or .endswith(".db")` and I'll cancel my PR #4695?

Reply:
That's easy to do, sure.

Reply:
Resolved.

> Can we add or `.endswith(".db")`

I only recently learned about this, but `.endswith`/`.startswith` can take tuples of suffixes/prefixes, so you don't have to manually do the `or`. `path.endswith((".sqlite", ".db"))` does it for you.